pipeline/hard-reset.sh

```diff
@@ -90,46 +91,46 @@ if [ "${force_yes}" = false ] ;then
 fi
 
 ######### functions ###########
-hard_reset_shard
+function hard_reset_shard
 {
    shard=$1
-   msg "cat $logdir/shard${shard}.txt | xargs -i{} -P50 bash -c \"./restart_node.sh -U -X -D -y -d logs/${net_profile} -p ${net_profile} -t 0 -r 0 -R 0 {}\""
-   cat "$logdir/shard${shard}.txt" | xargs -i{} -P50 bash -c "./restart_node.sh -U -X -D -y -d logs/${net_profile} -p ${net_profile} -t 0 -r 0 -R 0 {}"
+   msg "cat $logdir/shard${shard}.txt | xargs -i% -P50 bash -c \"./restart_node.sh -U -X -D -y -d logs/${net_profile} -p ${net_profile} -t 0 -r 0 -R 0 %\""
+   cat "$logdir/shard${shard}.txt" | xargs -i% -P50 bash -c "./restart_node.sh -U -X -D -y -d logs/${net_profile} -p ${net_profile} -t 0 -r 0 -R 0 %"
 }
 
-check_consensus
+function check_consensus
 {
    shard=$1
    msg "./run_on_shard.sh -p ${net_profile} -T $shard 'tac /home/tmp_log/*/zeorlog*.log | grep -m 1 HOORAY'"
 }
 
-check_version
+function check_version
 {
    shard=$1
```

Comment:
may NOT be necessary, I usually do `local shard=$1` to avoid variable contamination.